GitCommands/Git/GitRevision.cs

```diff
@@ -86,6 +82,77 @@ public BuildInfo BuildStatus
         public string Subject { get; set; } = "";
         [CanBeNull]
         public string Body { get; set; }
+
+        [CanBeNull]
+        public string BodySummary
+        {
+            get
+            {
+                if (Body == null || !HasMultiLineMessage)
+                {
+                    return Body;
+                }
+
+                return _bodySummary ?? (_bodySummary = BuildSummary(Body.TrimEnd()));
+            }
+        }
+
+        private const int MaxLineLength = 150;
+        private static string BuildSummary(string body)
+        {
+            var s = new StringBuilder(100);
+
+            int lineCount = 0;
+            int lineStartPos = 0;
+            int pos;
+            for (pos = 0; pos < body.Length; ++pos)
+            {
+                if (body[pos] == '\n')
+                {
+                    if (pos - lineStartPos > MaxLineLength)
+                    {
+                        AppendLine(MaxLineLength, withEllipsis: true);
+                    }
+                    else
+                    {
+                        AppendLine(pos - lineStartPos);
+                    }
+
+                    lineStartPos = pos + 1;
+
+                    if (++lineCount == 30)
```

Comment:
Please define a const, so we don't have random magic numbers throughout the code
Applies to other instances as well